pkg/apis/cdap/v1alpha1/cdapmaster_types.go

```diff
@@ -133,10 +133,21 @@ type UserInterfaceSpec struct {
 	CDAPExternalServiceSpec `json:",inline"`
 }
 
+// UpgradeJobSpec defines the specification for the upgrade job
+type UpgradeJobSpec struct {
```

Comment:
This is not a public type that is used by the user, right? Seems like it should be just a local struct in the controller